### ISSUE TITLE
Add a winninig effect in Memorize activity

### ISSUE THREAD

**Issue by jigyasunigam** [llaske/sugarizer#1018]
Add a winning cum congratulation message in the memorize activity after the game is over.

**Comment by Shreyas-SAS** [comment]
> Add a winning cum congratulation message in the memorize activity after the game is over.

Actually, I find it the case with every activity. whenever we clear a game we either get a very simple level cleared text on the top or absolutely nothing many a times. 

I feel that there is no such reward or excitement coming from the activity to actually appreciate the win. so even if you win you do not feel like you really did something and after maybe 5 or 6 rounds I should say, you don't feel like continuing anymore.

When the target audience is children, the too simple win message without any celebration or anything really gets boring and hard to continue. 

I really hope this gets changed in near future.

**Comment by llaske** [comment]
It seems obvious in the activity that the game is win.
Plus Sugarizer target children that can't read so a message is not pertinent.

**Comment by Shreyas-SAS** [comment]
> It seems obvious in the activity that the game is win. Plus Sugarizer target children that can't read so a message is not pertinent.

I would like to differ from the statement. As you said, sugarizer targets children that can't read at this stage, it becomes essential to keep moving objects and winning messages like a firework or some cartoon or something to actually attract the children. 

I do not wish to criticize in a wrong way but to point out from a progressive and positive point of view that the current system looks a bit off-targeted. When it comes to kids as niche audience using the software, I would say that the system looks a bit pale. those children who cannot read well, we cannot expect them to respect the concept of pair buddy color scheme. 

I fully understand that I stand in no position to even speak against the current standard and fully acknowledge the efforts put in for the software. I am really sorry if I unknowingly brought you a problem with this view. I am always bat at how I keep my view and I really apologize for it.

**Comment by jigyasunigam** [comment]
@llaske 
I totally agree with you. But can we add the spray of clapping emojis with a applause or yay sound when the game is over. This will motivate children to go on more harder level.

**Comment by Shreyas-SAS** [comment]
> @llaske I totally agree with you. But can we add the spray of clapping emojis with a applause or yay sound when the game is over. This will motivate children to go on more harder level.

exactly, if sir does not want to change the current system the I would really appreciate adding non visible elements like sound and music to the activities. It will improve a lot i believe. 

If @llaske sir is open to it I am willing to work on it.

**Comment by llaske** [comment]
Okay let's suggest a visual effect (not a message).
Caution: in multi-player mode the visual effect should be visible only for the winner.

**Comment by Shreyas-SAS** [comment]
> Okay let's suggest a visual effect (not a message).
> Caution: in multi-player mode the visual effect should be visible only for the winner.

Can we keep it like fireworks or butterflies flying around when you win. Although it is inspired from solitaire collection game which is pre-installed in windows I guess it will still serve the purpose. 

Else we can even show cartoons/teddies that will just dance on the screen when we win. At least that's what children love watching on the kids TV channels even if it does not make very much of a sense.

For background music we can just keep clapping or some short music no longer than 3 seconds in length. 

What do you say @llaske sir. Can we go ahead with this?

**Comment by jigyasunigam** [comment]
@Shreyas-SAS 
Yes Shreyas you can go with any visual effect just kept in mind that it should be for the children. As llaske told, instead of asking questions or sending messages give some impementations or send a pullrequest which resolves this issue.
"In short start working on this and next time share your work and it's images in the comment section. llaske will told you if your work is upto mark or it needs some improvement."

**Comment by Shreyas-SAS** [comment]
> @Shreyas-SAS Yes Shreyas you can go with any visual effect just kept in mind that it should be for the children. As llaske told, instead of asking questions or sending messages give some impementations or send a pullrequest which resolves this issue. "In short start working on this and next time share your work and it's images in the comment section. llaske will told you if your work is upto mark or it needs some improvement."

oh ok. I will first produce some work and then ask for approval and improvement suggestions. I just felt like I should take the permission first so that i do not go the wrong way as i gave three options as suggestion. It was really nice of you to allow me take up the issue you raised. 

If you want we can maybe do the same in two different activities too, like you can take up memorizer and i will take up some other. since the issue is common for many activities it will not be a problem.

@jigyasunigam btw which one do you suggest? fireworks, butterfly or cartoon dancing one? please do tell i will follow that one.

**Comment by jigyasunigam** [comment]
@Shreyas-SAS 
I think fireworks with applause sound will be the best. And yes you can start working on this, I don't want two persons working on this small issue .You can ask me about any suggestion or help in the near future, I will be happy to do it.
Remember : In multi-player mode the visual effect should be visible only for the winner.

**Comment by Shreyas-SAS** [comment]
> Remember : In multi-player mode the visual effect should be visible only for the winner.

Yeah that is my only concern for this task. I do not have a very extensive knowledge about managing things different for multiplayer and single player. 

Maybe you can complete this issue and i will see your code to understand the difference. That way i will get to know the handling and implement the same issue in other activities that lack the winning visuals.

@jigyasunigam are you fine with taking this task up? If you are busy then u will try but if you could help it would be really nice.

**Comment by jigyasunigam** [comment]
> are you fine with taking this task up? If you are busy then u will try but if you could help it would be really nice.
No problem I will do it.

**Comment by Shreyas-SAS** [comment]
> No problem I will do it.

Thanks. I will definitely study your code well and implement it on other activities having this issue.

**Comment by zadkhan** [comment]
Hello. It looks like this issue was never addressed. 
I am starting on this now - I plan to add fireworks with an applause noise in hopes to be inclusive for all sugarizer users.

**Comment by Shreyas-SAS** [comment]
@zadkhan sure, go ahead and propose a bit of work and designs to Lionel, if he likes it then as per previous conversation he might merge it in.

**Comment by AdityaKrSingh26** [comment]
Hey @llaske is this enhancement still able to be worked upon ??

**Comment by llaske** [comment]
@AdityaKrSingh26 I don't think it's useful. We can close this one.
